dashboard/modules/reporter/reporter_agent.py

```diff
@@ -391,10 +391,13 @@ async def MemoryProfiling(self, request, context):
         # add 1 second sleep for memray overhead
         await asyncio.sleep(duration + 1)
         success, output = await p.detach_profiler(pid)
+        warning = None if success else output
```

Comment:
in the events of detach_profile fails, we are going to double print both `output` and `warning`? 

How would that look like.

Reply:
the output will be the flamegraph of the profiler result, with additional warning on top telling that detach failed just to inform user that profiler still be attached to the process.
But if they run the profiler again, the next attach will automatically detach the previous profiler so it won't affect the next profiling result.

Reply:
if detach_profiler and get_profile_results both fails - don't you double print the error messages?

Reply:
no, we will just return the `get_profile_results` fails.

Reply:
so if `detach_profiler` fails - you have the warning 
and if get_profile_result fails - you have the errors in the output. 

And when you render the results later, you are adding them both? 
https://github.com/ray-project/ray/pull/41537/files/86145908cda9eb148d8ab70ccf797136f80be661..c46adb48d44f78384337c8fef13ef7a8b7a2d20a#diff-3540e3dc6b989a66f62c1d9e6de1cbe39f4db0eb957435fbeef8719fa161c6d2R604

Reply:
correct! since get_profile_result is one that generate result here despite detach success or failing